pkg/basicstation/cups/update_info.go

```diff
@@ -229,68 +235,75 @@ func (s *Server) UpdateInfo(c echo.Context) error {
 		gtw.Attributes = make(map[string]string)
 	}
 
-	if gtw.LBSLNSSecret == nil {
-		return errLNSCredentials.WithAttributes("gateway_uid", uid)
-	}
-
-	if s.requireExplicitEnable || gtw.Attributes[cupsAttribute] != "" {
+	if s.requireExplicitEnable && gtw.Attributes[cupsAttribute] != "" {
 		if cups, _ := strconv.ParseBool(gtw.Attributes[cupsAttribute]); !cups {
 			return errCUPSNotEnabled.WithAttributes("gateway_uid", uid)
 		}
 	}
 
 	res := UpdateInfoResponse{}
 
-	if gtw.Attributes[cupsURIAttribute] == "" {
-		gtw.Attributes[cupsURIAttribute] = req.CUPSURI
-	}
-	if s.allowCUPSURIUpdate && gtw.Attributes[cupsURIAttribute] != req.CUPSURI {
-		res.CUPSURI = gtw.Attributes[cupsURIAttribute]
+	_, host, port, err := parseAddress("https", req.CUPSURI)
+	if err != nil {
+		return errInvalidCUPSURI.WithAttributes("uri", req.CUPSURI)
 	}
+	cupsAddress := fmt.Sprintf("%s:%s", host, port)
+
+	if s.allowCUPSURIUpdate && gtw.TargetCUPSURI != "" && gtw.TargetCUPSURI != cupsAddress {
+		if gtw.TargetCUPSKey.Value == nil {
+			return errTargetCUPSCredentials.New()
+		}
+		logger := logger.WithField("cups_uri", gtw.TargetCUPSURI)
+		logger.Debug("Configure CUPS")
+		res.CUPSURI = gtw.TargetCUPSURI
 
-	if credentials := gtw.Attributes[cupsCredentialsAttribute]; credentials != "" {
-		logger.WithField("cups_uri", gtw.Attributes[cupsURIAttribute]).Debug("Getting trusted certificate for CUPS...")
-		cupsTrust, err := s.getTrust(gtw.Attributes[cupsURIAttribute])
+		logger.Debug("Get trusted certificate for CUPS")
+		cupsTrust, err := s.getTrust(gtw.TargetCUPSURI)
 		if err != nil {
 			return err
 		}
-		cupsCredentials, err := TokenCredentials(cupsTrust, credentials)
+		cupsCredentials, err := TokenCredentials(cupsTrust, string(gtw.TargetCUPSKey.Value))
```

Comment:
Maybe we could also use a flow to update (rotate) CUPS credentials?

Reply:
Good suggestion. Done.